Recent nightly shows no info in backtraces

Every since a recent nightly update, both distributed nightlies and locally built `rustc` produce backtraces where every level of the stack is represented _only_ by their memory address and the label `<unknown>`:

```
$ RUST_BACKTRACE=full rustc +devel file.rs -Ztreat-err-as-bug
error[E0573]: expected type, found function `foo`
  --> file.rs:10:17
   |
10 |     let _ = S::<foo()> { x: [u8; foo()] };
   |                 ^^^^^ not a type

thread 'rustc' panicked at 'aborting due to `-Z treat-err-as-bug=1`', src/librustc_errors/lib.rs:534:13
stack backtrace:
   0:        0x1115d3c84 - <unknown>
   1:        0x1115d1036 - <unknown>
   2:        0x1115d0dba - <unknown>
   3:        0x10f251322 - <unknown>
   4:        0x1115d1731 - <unknown>
   5:        0x11126f3b4 - <unknown>
   6:        0x11128a9b6 - <unknown>
   7:        0x111299fdc - <unknown>
   8:        0x10bbfcae9 - <unknown>
   9:        0x10b3e579d - <unknown>
  10:        0x10b333c10 - <unknown>
  11:        0x10b45b5b5 - <unknown>
  12:        0x10b43abe6 - <unknown>
  13:        0x10b330794 - <unknown>
  14:        0x10b39064d - <unknown>
  15:        0x10b38fa48 - <unknown>
  16:        0x10b391032 - <unknown>
  17:        0x10b4121db - <unknown>
  18:        0x10b0fa37e - <unknown>
  19:        0x10b178086 - <unknown>
  20:        0x10b167175 - <unknown>
  21:        0x10b175484 - <unknown>
  22:        0x10b11ce69 - <unknown>
  23:        0x1115f2cce - <unknown>
  24:        0x10b166408 - <unknown>
  25:        0x10b0d516b - <unknown>
  26:        0x1115b958d - <unknown>
  27:        0x1115f234d - <unknown>
  28:        0x1115e7bc8 - <unknown>
  29:     0x7fff784492ea - <unknown>
  30:     0x7fff7844c248 - <unknown>
query stack during panic:
end of query stack
```

Sadly I don't have a more specific repro case, but I _can_ consistently reproduce locally. Older distributed nightlies do not present this behavior.

--- a/src/libstd/Cargo.toml
+++ b/src/libstd/Cargo.toml
@@ -25,7 +25,7 @@ unwind = { path = "../libunwind" }
 hashbrown = { version = "0.4.0", features = ['rustc-dep-of-std'] }
 
 [dependencies.backtrace]
-version = "0.3.25"
+version = "0.3.29"
 default-features = false # don't use coresymbolication on OSX
 features = [
   "rustc-dep-of-std", # enable build support for integrating into libstd

